lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -666,36 +666,29 @@ finalizeModel(GeneratedCodeBasicInfo &GCBI,
   revng_check(Binary.verify(true));
 }
 
-static void
-combineCrossCallSites(MetaAddress EntryPC, FunctionSummary &Summary) {
+static void combineCrossCallSites(auto &CallSite, auto &Callee) {
   using namespace ABIAnalyses;
   using RegState = model::RegisterState::Values;
 
-  for (auto &[PC, CallSites] : Summary.ABIResults.CallSites) {
-    if (PC == EntryPC) {
-      for (auto &[FuncArg, CSArg] :
-           zipmap_range(Summary.ABIResults.ArgumentsRegisters,
-                        CallSites.ArgumentsRegisters)) {
-        auto *CSV = FuncArg == nullptr ? CSArg->first : FuncArg->first;
-        auto RSFArg = FuncArg == nullptr ? RegState::Maybe : FuncArg->second;
-        auto RSCSArg = CSArg == nullptr ? RegState::Maybe : CSArg->second;
+  for (auto &[FuncArg, CSArg] :
+       zipmap_range(Callee.ArgumentsRegisters, CallSite.ArgumentsRegisters)) {
+    auto *CSV = FuncArg == nullptr ? CSArg->first : FuncArg->first;
+    auto RSFArg = FuncArg == nullptr ? RegState::Maybe : FuncArg->second;
+    auto RSCSArg = CSArg == nullptr ? RegState::Maybe : CSArg->second;
```

Comment:
Can you elaborate (in the commit message) what type of improvement is going on here?
Were we not propagating information correctly across call sites?